### ISSUE TITLE
Confusing table

### ISSUE THREAD

**Issue by woctezuma** [Lusito/forget-me-not#182]
I find the table confusing, especially the part highlighted in yellow:
- what does ticking the "cleanup rule" do?

![confusing table](https://i.imgur.com/S7H9vAp.png)

Similarly, I find the symbols confusing:

- what does ticking the "garbage bin" do?
- what does clicking on the shield do?

![confusing symbols](https://i.imgur.com/PK2v4Cu.png)

**Comment by woctezuma** [comment]
What adds to the confusion is that the ticks are still visible if the "cleaning rule" is disabled.

![visible ticks](https://i.imgur.com/hHB8DjD.png)

After thinking a bit about it, I assume that the granularity allows the user to toggle the cleaning, and toggle the rules independently, so that, for instance, one is able to:

- deactivate cleaning at browser start,
- or deactivate the rules so that all the cookies are cleaned at browser start.

**Comment by Lusito** [comment]
I think you figured it out mostly on your own:
- The "Enable" is a global enable.. i.e. before you can cleanup cookies on startup, you need to enable cleanup on startup. I can see this being confusing if the disabled checkboxes are still ticked..
- Upon manual cleanup, ticking the garbage bin means running cleanup in that category.
- If the shield has a red line over it, it means rules are not being applied. => All data is being removed, regardless of your rules.
- Cleaning "Downloads" is named a bit confusing and should probably be renamed to "Download History". All rules can be applied to anything that has a URL. A download history item has a URL, so those items can be affected by the rules.

From your last paragraph, I'm assuming you noticed a bug during download cleanup. There are actually two places where downloads appear: The regular history window and the list of downloads (toolbar download icon).

Could you write down a list of steps that you did and also write down if the downloads have (not) been cleared in regular history window and list of downloads?

Thanks

**Comment by woctezuma** [comment]
>     * The "Enable" is a global enable.. i.e. before you can cleanup cookies on startup, you need to enable cleanup on startup. I can see this being confusing if the disabled checkboxes are still ticked..

Right. It would be clearer if the boxes were all unticked instead (or on top) of having this overlay.

![visible ticks](https://i.imgur.com/hHB8DjD.png)

> Could you write down a list of steps that you did and also write down if the downloads have (not) been cleared in regular history window and list of downloads?

In my previous browsing session, I have downloaded the installer for the Epic Games store.
During this session, I have downloaded the installer for Git for Windows.

My settings are:

If I look at the list of downloads, I see:
![toolbar](https://i.imgur.com/ywkhBFD.png)

If I press the button to see the whole download history (or alternatively press <Ctrl+J>), I see:
![whole history](https://i.imgur.com/V4W1iBY.png)

So, it seems that the download history was not flushed when the browser was restarted.

**Comment by Lusito** [comment]
Technically it is a bug, but one I can't solve at this time, as for some reason, firefox clears the downloads list upon shutdown, yet keeps the download history, without me being able to detect if the history entry is a download or not. That's why the add-on remembers which downloads have been made in order to clean them.

Thanks for testing this.

**Comment by woctezuma** [comment]
> Cleaning "Downloads" is named a bit confusing and should probably be renamed to "Download History". All rules can be applied to anything that has a URL. A download history item has a URL, so those items can be affected by the rules.

I forgot to reply to this part.

It would be clearer to mention the "download history" rather than "downloads", especially because there is a setting about local files, which I also find confusing.

![local files](https://i.imgur.com/zbe7lGu.png)

**Comment by MrQwerky** [comment]
When the **Enable** box is _unchecked_, the remaining rule boxes remain checked:  _This is good!_  In my opinion, it is very good to be able to see what the settings are which will take effect once the **Enable** box becomes checked.  Please do not change to hide the checked boxes when **Enable** is unchecked.  I hate Microsoft's (and Mozilla's) attempts to hide everything under the sun, which makes the user work harder just to see what the settings are.  JMO.
